### DISCUSSION TITLE
Running in native mode results in a blank window.

### DISCUSSION THREAD

**Discussion by alimate2023** [zauberzeug/nicegui#2737] [General]
Running in native mode results in a blank window.
The console displays a warning message as follows
I don't know if elements are not displayed in the native window.

```py
from nicegui import native, ui
from nicegui.elements import markdown

ui.icon('thumb_up')
ui.markdown('This is **Markdown**.')
ui.html('This is <strong>HTML</strong>.')
with ui.row():
    ui.label('CSS').style('color: #888; font-weight: bold')
    ui.label('Tailwind').classes('font-serif')
    ui.label('Quasar').classes('q-ml-xl')
ui.link('NiceGUI on GitHub', 'https://github.com/zauberzeug/nicegui')

# ui.run()

ui.label('Hello from PyInstaller')

# ui.run(reload=False, port=native.find_open_port())
ui.run(reload=False,native=True, window_size=(400, 300), fullscreen=False)
```

## console error
NiceGUI ready to go on http://localhost:8000
[pywebview] MSHTML is deprecated. See https://pywebview.flowrl.com/guide/renderer.html#web-engine on details how to 
use Edge Chromium

![캡처](https://github.com/zauberzeug/nicegui/assets/151822282/6ea7bb67-42bf-4198-88a4-462f58c58246)

**Comment by falkoschindler** [comment]
Hi @alimate2023,

Have you installed the newest versions of NiceGUI and pywebview?
And can you, please, show the full console log after starting the app? There's a KeyboardInterrupt in your screenshot - where is this coming from?
Thanks!

**Reply by alimate2023** [reply]
```
PS C:\Data\Project\alimate2_nicegui> pip freeze
aiofiles==23.2.1
aiohttp==3.9.3
aiosignal==1.3.1
annotated-types==0.6.0
anyio==4.3.0
attrs==23.2.0
beautifulsoup4==4.12.3
bidict==0.23.1
bottle==0.12.25
bs4==0.0.2
CacheControl==0.14.0
cachetools==5.3.3
certifi==2024.2.2
cffi==1.16.0
charset-normalizer==3.3.2
click==8.1.7
clr-loader==0.2.6
colorama==0.4.6
cryptography==42.0.5
docutils==0.19
fastapi==0.109.2
firebase-admin==6.5.0
frozenlist==1.4.1
google-api-core==2.17.1
google-api-python-client==2.122.0
google-auth==2.29.0
google-auth-httplib2==0.2.0
google-cloud-core==2.4.1
google-cloud-firestore==2.15.0
google-cloud-storage==2.16.0
google-crc32c==1.5.0
google-resumable-media==2.7.0
googleapis-common-protos==1.63.0
grpcio==1.62.1
grpcio-status==1.62.1
h11==0.14.0
httpcore==1.0.4
httplib2==0.22.0
httptools==0.6.1
httpx==0.27.0
idna==3.6
ifaddr==0.2.0
itsdangerous==2.1.2
Jinja2==3.1.3
loguru==0.7.2
markdown2==2.4.13
MarkupSafe==2.1.5
msgpack==1.0.8
multidict==6.0.5
nicegui==1.4.18
numpy==1.26.4
orjson==3.9.15
outcome==1.3.0.post0
pandas==2.2.1
proto-plus==1.23.0
protobuf==4.25.3
proxy_tools==0.1.0
pscript==0.7.7
pyasn1==0.5.1
pyasn1-modules==0.3.0
pycparser==2.21
pydantic==2.6.4
pydantic_core==2.16.3
Pygments==2.17.2
PyJWT==2.8.0
pyparsing==3.1.2
PySocks==1.7.1
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
python-engineio==4.9.0
python-multipart==0.0.9
python-socketio==5.11.1
pythonnet==3.0.3
pytz==2024.1
pywebview==5.0.5
PyYAML==6.0.1
requests==2.31.0
rsa==4.9
selenium==4.18.1
simple-websocket==1.0.0
six==1.16.0
sniffio==1.3.1
sortedcontainers==2.4.0
soupsieve==2.5
starlette==0.36.3
trio==0.25.0
trio-websocket==0.11.1
typing_extensions==4.10.0
tzdata==2024.1
uritemplate==4.1.1
urllib3==2.2.1
uvicorn==0.29.0
vbuild==0.8.2
watchfiles==0.21.0
websockets==12.0
win32-setctime==1.1.0
wsproto==1.2.0
yarl==1.9.4
```

**Reply by alimate2023** [reply]
![캡처2](https://github.com/zauberzeug/nicegui/assets/151822282/3b005a52-2071-407e-a10c-378dd487c1eb)

**Reply by alimate2023** [reply]
Only a blank screen appears as if you can see it.

**Comment by alimate2023** [comment]
somebody help me! plz ..

**Comment by rodja** [comment]
@alimate2023 thanks for providing more details. Instead of screenshots, text would have been much better because it would allow us to copy paste code or the url. Have you read the link behind the url in the pywebview warning?

**Reply by alimate2023** [reply]
https://pywebview.flowrl.com/guide/renderer.html#web-engine
I can't access this url.

**Reply by rodja** [reply]
To bad. But then, querying a search engine could help. It came up with some instructions about [engine selection in pywebview](https://pywebview.flowrl.com/3.7/guide/renderer.html#:~:text=Chrome%2066-,On%20Windows%20renderer%20is%20chosen%20in%20the%20following%20order%3A%20edgechromium,.,-To%20change%20a). In either case I more suspect this to be a pywebview problem than a NiceGUI one...

**Reply by alimate2023** [reply]
thanks. rodja!
It's a webview problem, I was wondering how to set it up in nicegui.

**Reply by falkoschindler** [reply]
I thought the pywebview warning might be unrelated, because even if something is deprecated, it should still work. But I'm not sure.

**Reply by alimate2023** [reply]
@rodja @falkoschindler 
Does it show normally if I run it with my code?

**Reply by falkoschindler** [reply]
@alimate2023 It runs normally on our machines, which run macOS. And there's no deprecation either.

**Reply by alimate2023** [reply]
@falkoschindler 
I'll do it in the old MacBook
 :-)

**Reply by alimate2023** [reply]
![image](https://github.com/zauberzeug/nicegui/assets/151822282/69133f60-18e5-41db-b426-5d1cd430d1b7)
If you simply run pywebview, the screen will display normally.
However, it is not visible in nicegui.

**Reply by alimate2023** [reply]
@rodja @falkoschindler 
Please test it on the Windows 11 OS.

**Reply by falkoschindler** [reply]
Sorry, @alimate2023, we don't have access to Windows machines at the moment. I just created issue https://github.com/zauberzeug/nicegui/issues/2751 from this discussion. Maybe someone from the community can help.

**Reply by gsal** [reply]
I just run into this same issue.

Linux Rocky 9
Python 3.9
Brand new python virtual environment, 
pip installed nicegui, qt, pywebview[qt]
Nicegui window is blank
BUT, not just nicegui
if I run the (non-nicegui)  3-line test3.py webview program above, I also just get  blank screen and no error messages at all.
what gives?

**Reply by gsal** [reply]
Ok, started over; this time with a conda envionment. The test3.py above works.

**Comment by Muyv** [comment]
check your system proxy

**Reply by python-and-novella** [reply]
Have you seen the init problem?

**Reply by Muyv** [reply]
try to use pywebview create window with `http://localhost:8000` (nicegui app), start with arg `debug=True`, it will shows developer's tool, then you will see the errors.

**Reply by python-and-novella** [reply]
the problem is MSHTML , not system proxy,don't mistake it.

**Reply by Muyv** [reply]
not the same problem I had, thank you for providing more useful reference

**Comment by python-and-novella** [comment]
@alimate2023
Plz try to set gui mode to ‘edgechromium’ (need to install webview2 on windows) or 'qt'\'pyside2'\'pyside6' (need to install dependencies , check it in pywebview's doc).
default ‘mshtml’ is IE, that is not compatible with nicegui.
see it in https://pywebview.flowrl.com/guide/web_engine.html#web-engine
```python3
from nicegui import  ui,app

app.native.start_args['gui'] = 'edgechromium' 
ui.icon('thumb_up')
ui.markdown('This is **Markdown**.')
ui.html('This is <strong>HTML</strong>.')
with ui.row():
    ui.label('CSS').style('color: #888; font-weight: bold')
    ui.label('Tailwind').classes('font-serif')
    ui.label('Quasar').classes('q-ml-xl')
ui.link('NiceGUI on GitHub', 'https://github.com/zauberzeug/nicegui')

ui.label('Hello from PyInstaller')

ui.run(reload=False,native=True, window_size=(400, 300), fullscreen=False)
```
